.expeditor/scripts/finish_release/bump_rustfmt.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+# This script attempts to find the latest installable nightly rustfmt.
+# When found, it bumps the repo rustfmt version and submits a new PR
+# with the new version pin and any code base formatting changes from the
+# new version.
+
+# Assumptions:
+# 1. This script runs on a x86_64-unknown-linux-gnu target
+# 2. If rustfmt is installable on x86_64-unknown-linux-gnu, the other Tier1
+#    platforms are assumed to also have a working version.
+
+set -Eeuo pipefail
+
+# shellcheck source=.expeditor/scripts/shared.sh
+source .expeditor/scripts/shared.sh
+
+TODAY=$(date '+%Y-%m-%d')
+# pick out the date from a string like: nightly-2019-12-04
+CURRENT_RUSTFMT_DATE=$(awk -F- '{print $2"-"$3"-"$4}' RUSTFMT_VERSION)
+# convert to seconds for easier date comparison later
+CURRENT_RUSTFMT_DATE_S=$(date '+%s' --date "${CURRENT_RUSTFMT_DATE}")
+CHANNEL=nightly
+# a maximum iterator value for searching elligible nightlies
+MAX_ATTEMPTS=90
+
+# if set, this will be the next version
+export FOUND_VERSION
+
+find_nightly_rustfmt() {
+  # set minimal profile so only rustc, rust-std, and cargo components
+  # are installed in addition to rustfmt
+  echo "--- :habicat: Setting minimal profile"
+  rustup set profile minimal
+
+  for ((i=0; i<MAX_ATTEMPTS; i++)); do
+    # begin searching nightly date versions using current date
+    # as a starting point
+    attempt_date=$(date '+%Y-%m-%d' --date "${TODAY} -${i} day")
+    # use the conversion to seconds for easy date comparison
+    attempt_date_s=$(date '+%s' --date "${attempt_date}")
+    attempt_version="${CHANNEL}-${attempt_date}"
+    echo "--- Trying Nightly Date Version: ${attempt_version}"
+    if rustup toolchain install "${attempt_version}" --component rustfmt; then
+      FOUND_VERSION=${attempt_version}
+      break
+    fi
+    if [ "${attempt_date_s}" -le "${CURRENT_RUSTFMT_DATE_S}" ]; then
+      # Exit early since there is no newer version available than what is
+      # currently pinned in RUSTFMT_VERSION file.
+      echo "--- No newer ${CHANNEL} versions found. Nothing to do."
+      exit 0
+    fi
+  done
+}
+
+run_fmt_open_pr() {
+  local version_underbar="${FOUND_VERSION//-/_}"
+
+  # Do not change the branch name here without also changing
+  # it in .expeditor/config.yml where a workload
+  # subscription references the same name.
+  readonly branch="expeditor/rustfmt_${version_underbar}"
+  maybe_run git checkout -b "${branch}"
```

Comment:
I wonder if we really need the `maybe_run` safety rails in this case. The script isn't doing anything that's critical or irreversable (the worst it could do is create a PR and send a Slack message).

Without `maybe_run` on these actions, we could actually try running this for real before the next release (potentially fixing any bugs in the meantime), and could potentially have more flexibility for when we want to run it in the future.

(Not a huge deal either way, but just an observation.)

Reply:
I feel there's a bit of irony here since your reasons are the exact reasons that I chose to add `maybe_run` to these items 😄 It allows me to test run the script in the `chefes/lita-worker` docker image and iron out bugs. If I remove that, then it

* checks out and commits my WIP in a different branch 🤦‍♂ 
* dies when calling `post_slack_message` function 😞 

With it, we can test like so:

```
✔ ~/Projects/habitat [jeremymv2/auto_bump_rustfmt L|✚ 1…1⚑ 4]
13:15 $ docker run -it --rm --init --privileged --volume ${PWD}:${PWD} --env BUILDKITE_BUILD_CREATOR='' --env EXPEDITOR_SCM_SETTINGS_BRANCH='jeremymv2/auto_bump_rustfmt' --env GITHUB_TOKEN='A_TOKEN' --workdir ${PWD} chefes/lita-worker bash -c ".expeditor/scripts/finish_release/notify_rustfmt.sh"

--- :github: Installing hub version v2.13.0 to /bin/hub from https://github.com/github/hub/releases/download/v2.13.0/hub-linux-amd64-2.13.0.tgz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   614    0   614    0     0   2882      0 --:--:-- --:--:-- --:--:--  2882
100 4975k  100 4975k    0     0  5778k      0 --:--:-- --:--:-- --:--:-- 16.3M
--- :github: Fetching PR URL of jeremymv2/auto_bump_rustfmt branch
--- :slack: Notifying hab-team
Build initiated by
Would have run: post_slack_message hab-team "Hi @habitat-team, there is a rustfmt version bump PR ready for review! :jumping-habicat:

https://github.com/habitat-sh/habitat/pull/7258"

Reply:
Ah! I guess I need to read up on local testing of this stuff with `chefes/lita-worker`!

I was coming at it from a "run this in Buildkite for real" perspective, but yours is also 100% valid.

I'm happy to leave it as-is, for sure. As a thinking exercise, I wonder if there's some way we could approach things in the future that would maybe allow us to do both? :thinking:

Reply:
I think there is definitely room for improvement in creating testable pipeline steps/stages without overstepping a "sandbox".

As a point of reference, I did learn about the lita-work image here: https://expeditor.chef.io/docs/troubleshooting/debugging-bash-scripts/